app/components/agenda-manager/index.js

```diff
@@ -13,35 +11,20 @@ import {task} from "ember-concurrency-decorators";
 
 /** @extends {Component<Args>} */
 export default class AgendaManagerIndexComponent extends Component {
-  /** @type AgendaData */
-  @tracked popup = false;
-  @tracked editModalVisible = false;
   @tracked itemToEdit = null;
 
-  constructor(...args) {
-    super(...args);
-  }
-
-
-
-  @action
-  openModal() {
-    this.editModalVisible = true;
-  }
-
-  @action
-  closeModal() {
-    this.editModalVisible = false;
+  get editModalVisible() {
+    return Boolean(this.itemToEdit);
   }
 
   @task
   * createItemTask(newItemTask) {
     const newItem = yield newItemTask.perform();
     this.editItem(newItem);
   }
+
   @action
   cancelEdit() {
```

Comment:
`cancelEdit` isn't the best name anymore since it's now also used for simply closing the modal. `stopEditing` might be clearer?

Reply:
I think on closing the modal, the expectation is to also rollback changes. at least that's how the other modals work afaik. (in essence the close button should behave as a second cancel button).

Reply:
I think that's how it works now? I'll have to double check how I implemented it though. I have the memory of a fish :smile:.

Reply:
Just verified. `cancelEdit` is passed around to the nested components and used after other actions are complete:
- after saving
- after deletion
- after resetting

The reset functionality gets called every time the modal needs to be closed without using the save or delete buttons.

It's only used to close the modal, which is why the name isn't the best at the moment. I can just rename it to `closeModal` maybe, but it's a bit weird since the code in the action only clears the `itemToEdit` prop (which is why I suggested `stopEditing` I think).

Reply:
Renamed to `stopEditing`